linker/Mono.Linker/Driver.cs

```diff
@@ -294,6 +296,7 @@ static void Usage (string msg)
 			Console.WriteLine ("                 default is all");
 			Console.WriteLine ("   -x          Link from an XML descriptor");
 			Console.WriteLine ("   -a          Link from a list of assemblies");
+			Console.WriteLine ("   -r          Link from a list of assemblies using public roots");
```

Comment:
I made several changes to the PR:
1. Family and FamilyOrAssembly methods and fields are marked. That addresses your comment about protected fields/methods. Also, this keeps destructors since they get emitted as protected Finalize methods.
2. NestedFamily and NestedFamilyOrAssembly nested classes are marked. Public nested classes from internal enclosing classes are still not marked since they can't be accessed from other assemblies.
3. If assembly has InternalsVisibleTo attribute, then internal classes, fields, and methods are marked.